lib/ansible/modules/crypto/luks_device.py

```diff
@@ -0,0 +1,460 @@
+
+    def run_luks_close(self, name):
+        result = self._run_command([self._cryptsetup_bin, 'close', name])
+        if result[RETURN_CODE] != 0:
+            raise ValueError('Error while closing LUKS container %s' % (name))
+
+    def run_luks_remove(self, device):
+        wipefs_bin = self._module.get_bin_path('wipefs', True)
+
+        name = self.get_container_name_by_device(device)
+        if name is not None:
+            self.run_luks_close(name)
+        result = self._run_command([wipefs_bin, '--all', device])
+        if result[RETURN_CODE] != 0:
+            raise ValueError('Error while wiping luks container %s: %s'
+                             % (device, result[STDERR]))
+
+    def run_luks_add_key(self, device, keyfile, new_keyfile):
+        ''' Add new key to given 'device'; authentization done using 'keyfile'
+            Raises ValueError when command fails
+        '''
+        result = self._run_command([self._cryptsetup_bin, 'luksAddKey', device,
+                                    new_keyfile, '--key-file', keyfile])
+        if result[RETURN_CODE] != 0:
+            raise ValueError('Error while adding new LUKS key to %s: %s'
+                             % (device, result[STDERR]))
+
+    def run_luks_remove_key(self, device, keyfile):
+        ''' Remove key from given device
+            Raises ValueError when command fails
+        '''
+        result = self._run_command([self._cryptsetup_bin, 'luksRemoveKey', device,
+                                    '-q', '--key-file', keyfile])
+        if result[RETURN_CODE] != 0:
+            raise ValueError('Error while removing LUKS key from %s: %s'
+                             % (device, result[STDERR]))
+
+
+class ConditionsHandler(Handler):
+
+    def __init__(self, module, crypthandler):
+        super(ConditionsHandler, self).__init__(module)
+        self._crypthandler = crypthandler
+
+    def luks_create(self):
+        return (self._module.params['device'] is not None and
+                self._module.params['keyfile'] is not None and
+                self._module.params['state'] == 'present' and
+                not self._crypthandler.is_luks(self._module.params['device']))
+
+    def luks_open(self):
+        if (self._module.params['device'] is None or
+                self._module.params['keyfile'] is None or
+                self._module.params['open'] is False):
+            # conditions for open not fulfilled
+            return False
+
+        if self._module.params['state'] == 'absent':
+            # better warn the user to recheck the configuration
+            self._module.fail_json(msg="Contradiction in setup: LUKS set "
+                                   "to be 'absent' and 'open'.")
+
+        # try to obtain luks name - it may be already opened
+        name = self._crypthandler.get_container_name_by_device(
+            self._module.params['device'])
+        if name is not None:
+            if name != self._module.params['name']:
```

Comment:
If `self._module.params['name'] is None`, this will never match and the module fails.

Reply:
Fixed.